include/ozo/connection_pool.h

```diff
@@ -70,68 +99,51 @@ struct pooled_connection_wrapper {
     }
 };
 
-template <typename Connector, typename OidMap, typename Statistics, typename Handler>
-auto wrap_pooled_connection_handler(io_context& io, Connector&& connector,
-    Statistics&& stat, Handler&& handler) {
-    return pooled_connection_wrapper<std::decay_t<Connector>, OidMap,
-            std::decay_t<Statistics>, std::decay_t<Handler>>{
-        io, std::forward<Connector>(connector), std::forward<Statistics>(stat),
-        std::forward<Handler>(handler)};
+template <typename P, typename Handler>
+auto wrap_pooled_connection_handler(P&& provider, Handler&& handler) {
+
+    static_assert(ConnectionProvider<P>, "is not a ConnectionProvider");
+
+    return pooled_connection_wrapper<std::decay_t<P>, std::decay_t<Handler>> {
+        std::forward<P>(provider), std::forward<Handler>(handler)
+    };
 }
 
 } // namespace impl
 
-struct async_connector {
-    std::string conn_info_;
-    template <typename T, typename Handler>
-    Require<Connectable<T>> operator()(T&& conn, Handler&& handler) const {
-        async_connect(conn_info_, std::forward<T>(conn),
-            std::forward<Handler>(handler));
-    }
-};
-
-// struct async_timed_connector {
-//     std::string conn_info_;
-//     std::chrono::steady_clock::duration timeout_;
-//     template <typename T, typename Handler>
-//     Require<Connectable<T>> operator()(T&& conn, Handler&& handler) const {
-//         decltype(auto) socket = get_socket(conn);
-//         async_connect(conn_info_, std::forward<T>(conn),
-//             with_timeout(socket, timeout_, std::forward<Handler>(handler)));
-//     }
-// };
-
-static_assert(Connectable<impl::pooled_connection_ptr<empty_oid_map, no_statistics>>,
+static_assert(Connectable<impl::pooled_connection_ptr<impl::connection<empty_oid_map, no_statistics>>>,
     "pooled_connection_ptr is not a Connectable concept");
 
-static_assert(ConnectionProvider<impl::pooled_connection_ptr<empty_oid_map, no_statistics>>,
+static_assert(ConnectionProvider<impl::pooled_connection_ptr<impl::connection<empty_oid_map, no_statistics>>>,
     "pooled_connection_ptr is not a ConnectionProvider concept");
 
-template <typename Connector, typename OidMap, typename Statistics>
+struct connection_pool_config {
+    std::size_t capacity = 1;
+    std::size_t queue_capacity = 1;
```

Comment:
Maybe we should put here more reasonable defaults. Because this looks like very restricted case. You can have only one connection and have only one pending request. This will not work in any of ours usual cases. I think 10 and 100 will be more useful.

Reply:
**Done**